torch/testing/_comparison.py

```diff
@@ -602,7 +602,9 @@ def _to_tensor(self, tensor_like: Any) -> torch.Tensor:
             raise UnsupportedInputs()
 
     def _check_supported(self, tensor: torch.Tensor, *, id: Tuple[Any, ...]) -> None:
-        if tensor.layout not in {torch.strided, torch.sparse_coo, torch.sparse_csr}:  # type: ignore[attr-defined]
+        supported_layouts = {torch.strided, torch.sparse_coo, torch.sparse_csr, torch.sparse_csc,
+                             torch.sparse_bsr, torch.sparse_bsc}
+        if tensor.layout not in supported_layouts:  # type: ignore[attr-defined]
```

Comment:
Given that we don't use `supported_layouts` anywhere else, can we remove it?

```suggestion
        if tensor.layout not in {
            torch.strided,
            torch.sparse_coo,
            torch.sparse_csr,
            torch.sparse_csc,
            torch.sparse_bsr,
            torch.sparse_bsc,
        }:
```

I've omitted the `type: ignore` comment here. Please only re-add it if `mypy` complains.

Reply:
Done.